language/compiler/ir-to-bytecode/syntax/src/syntax.rs

```diff
@@ -1 +1,1798 @@
+                )?;
+                fmt_expected(f, expected)
+            }
+        }
+    }
+}
+
+fn consume_token<'input, 'builder>(tokens: &mut __Matcher<'input, 'builder>, tok: Tok)
+    -> Result<(), ParseError<usize, Token<'input>, failure::Error>>
+{
+    if (tokens.token.1).0 != tok {
+        return Err(ParseError::InvalidToken { location: tokens.token.0 });
+    }
+    tokens.advance()?;
+    Ok(())
+}
+
+fn parse_name<'input, 'builder>(tokens: &mut __Matcher<'input, 'builder>)
+    -> Result<String, ParseError<usize, Token<'input>, failure::Error>>
+{
+    if (tokens.token.1).0 != Tok::NameValue {
+        return Err(ParseError::InvalidToken { location: tokens.token.0 });
+    }
+    let name = (tokens.token.1).1.to_string();
+    tokens.advance()?;
+    Ok(name)
+}
+
+fn parse_name_begin_ty<'input, 'builder>(tokens: &mut __Matcher<'input, 'builder>)
+    -> Result<String, ParseError<usize, Token<'input>, failure::Error>>
+{
+    if (tokens.token.1).0 != Tok::NameBeginTyValue {
+        return Err(ParseError::InvalidToken { location: tokens.token.0 });
+    }
+    let s = (tokens.token.1).1;
+    tokens.advance()?;
+    // The token includes a "<" at the end, so chop that off to get the name.
+    Ok(s[..s.len() - 1].to_string())
+}
+
+fn parse_dot_name<'input, 'builder>(tokens: &mut __Matcher<'input, 'builder>)
+    -> Result<String, ParseError<usize, Token<'input>, failure::Error>>
+{
+    if (tokens.token.1).0 != Tok::DotNameValue {
+        return Err(ParseError::InvalidToken { location: tokens.token.0 });
+    }
+    let name = (tokens.token.1).1.to_string();
+    tokens.advance()?;
+    Ok(name)
+}
+
+// AccountAddress: AccountAddress = {
+//     < s: r"0[xX][0-9a-fA-F]+" > => { ... }
+// };
+
+fn parse_account_address<'input, 'builder>(tokens: &mut __Matcher<'input, 'builder>)
+    -> Result<AccountAddress, ParseError<usize, Token<'input>, failure::Error>>
+{
+    if (tokens.token.1).0 != Tok::AccountAddressValue {
+        return Err(ParseError::InvalidToken { location: tokens.token.0 });
+    }
+    let addr = AccountAddress::from_hex_literal(&(tokens.token.1).1)
+        .unwrap_or_else(|_| panic!("The address {:?} is of invalid length. Addresses are at most 32-bytes long", (tokens.token.1).1));
+    tokens.advance()?;
+    Ok(addr)
+}
+
+// Var: Var = {
+//     <n:Name> =>? Var::parse(n),
+// };
+
+fn parse_var<'input, 'builder>(tokens: &mut __Matcher<'input, 'builder>)
+    -> Result<Var, ParseError<usize, Token<'input>, failure::Error>>
+{
+    Var::parse(parse_name(tokens)?)
+}
+
+fn parse_var_<'input, 'builder>(tokens: &mut __Matcher<'input, 'builder>)
+    -> Result<Var_, ParseError<usize, Token<'input>, failure::Error>>
+{
+    let v_start_loc = tokens.token.0;
+    let var = parse_var(tokens)?;
+    let v_end_loc = tokens.token.0;
+    Ok(Spanned{span: Span::new(ByteIndex(v_start_loc as u32),
+                               ByteIndex(v_end_loc as u32)),
+               value: var })
```

Comment:
Is `rustfmt`/ `cargo fmt` still ignored on this file? Spacing looks a bit weird

Reply:
Good catch. I wondered why clippy and "cargo fmt" seemed strangely quiet. Halfway through my development of this PR, the syntax.rs file was added git with fmt and clippy disabled. I've fixed that now.